stdlib/regex.gr

```diff
@@ -378,17 +402,17 @@ enum ParsedRegularExpression {
   RELiteral(Char),
   RELiteralString(String), // <- sequences of literals are flattened into a string
   REAlts(ParsedRegularExpression, ParsedRegularExpression),
-  RESequence(List<ParsedRegularExpression>, Bool), // seq elts, needs backtrack
-  REGroup(ParsedRegularExpression, Number), // regex, group ID
-  RERepeat(ParsedRegularExpression, Number, Option<Number>, Bool), // regex, min, max (None for infinity), true=non-greedy
-  REMaybe(ParsedRegularExpression, Bool), // regex, true=non-greedy
-  REConditional(ParsedRegularExpression, ParsedRegularExpression, Option<ParsedRegularExpression>, Number, Number, Bool), // test, if-true, if-false, n-start, num-n, needs-backtrack
-  RELookahead(ParsedRegularExpression, Bool, Number, Number), // regex, is-match, n-start, num-n
-  RELookbehind(ParsedRegularExpression, Bool, Box<Number>, Box<Number>, Number, Number), // regex, is-match, lb-min, lb-max, n-start, num-n (lb-xx values patched in later)
-  RECut(ParsedRegularExpression, Number, Number, Bool), // regex, n-start, num-n, needs-backtrack
-  REReference(Number, Bool), // n, case-sensitive
-  RERange(List<(Number, Number)>),
-  REUnicodeCategories(List<UnicodeCategory>, Bool) // symlist, true=match/false=does-not-match
+  RESequence(SeqElts, NeedsBackTrack), // seq elts, needs backtrack
+  REGroup(ParsedRegularExpression, GroupID),
+  RERepeat(ParsedRegularExpression, Min, Max, Greedy), // regex, min, max (None for infinity), true=non-greedy
+  REMaybe(ParsedRegularExpression, Greedy), // regex, true=non-greedy
```

Comment:
```suggestion
  RERepeat(ParsedRegularExpression, Min, Max, NonGreedy), // regex, min, max (None for infinity), true=non-greedy
  REMaybe(ParsedRegularExpression, NonGreedy), // regex, true=non-greedy
```
The comment says `true=non-greedy`, so I believe this should be called NonGreedy?